eskipfile/remote_test.go

```diff
@@ -195,3 +196,120 @@ func createTestServer(c string, sc int) *httptest.Server {
 		io.WriteString(w, c)
 	}))
 }
+
+func TestRoutesCaching(t *testing.T) {
+	var expected = []*eskip.Route{{
+		Id:   "VALID",
+		Path: "/",
+		Filters: []*eskip.Filter{{
+			Name: "setPath",
+			Args: []interface{}{
+				"/homepage/",
+			},
+		}},
+		BackendType: eskip.NetworkBackend,
+		Shunt:       false,
+		Backend:     "https://example.com/",
+	}}
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("ETag", "test-etag")
+		if noneMatch := r.Header.Get("If-None-Match"); noneMatch == "test-etag" {
+			print("\n============= I'm matched here =============\n")
+			// w.Header().Set("ETag", "test-etag")
+			w.WriteHeader(http.StatusNotModified)
+			return
+		}
+		io.WriteString(w, fmt.Sprintf("VALID: %v;", routeBody))
+	}))
+	defer server.Close()
+
+	options := &RemoteWatchOptions{RemoteFile: server.URL, Threshold: 10, Verbose: true, FailOnStartup: true}
+	client, err := RemoteWatch(options) // first call
+	if err == nil {
+		defer client.(*remoteEskipFile).Close()
+	}
+
+	_, err = client.(*remoteEskipFile).getRemoteData() // cached, to make sure it returns the right status code
+
+	if !errors.Is(err, errContentNotChanged) {
+		t.Error(err)
+		return
+	}
+
+	r, err := client.LoadAll() // cached, to make sure it returns the right content
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	if !eskip.EqLists(r, expected) {
+		t.Errorf("invalid cached routes received\n%s", cmp.Diff(r, expected))
+		return
+	}
+
+}
+
+func TestRoutesCachingWrongEtag(t *testing.T) {
+	var flushCache = false
```

Comment:
flushCache := false

Reply:
This variable is accessed by the test and test server so there is a race condition and the test likely will fail when launched with -race flag

Reply:
Consider using https://pkg.go.dev/sync/atomic#Bool